src/platform/android/AndroidChipPlatform-JNI.cpp

```diff
@@ -274,6 +277,63 @@ JNI_LOGGING_METHOD(void, setLogFilter)(JNIEnv * env, jclass clazz, jint level)
     SetLogFilter(category);
 }
 
+static void ENFORCE_FORMAT(3, 0) logRedirectCallback(const char * module, uint8_t category, const char * msg, va_list args)
+{
+    using namespace chip::Logging;
+
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+    int priority = ANDROID_LOG_DEBUG;
+    switch (category)
+    {
+    case kLogCategory_Error:
+        priority = ANDROID_LOG_ERROR;
+        break;
+    case kLogCategory_Progress:
+        priority = ANDROID_LOG_INFO;
+        break;
+    case kLogCategory_Detail:
+        priority = ANDROID_LOG_DEBUG;
+        break;
+    default:
+        break;
+    }
+
+    jstring jModule = env->NewStringUTF(module);
+    jint jPriority  = static_cast<jint>(priority);
+
+    char buffer[CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE];
+    vsnprintf(buffer, sizeof(buffer), msg, args);
+    jstring jMsg = env->NewStringUTF(buffer);
```

Comment:
Handle the case where jMsg is null due to UTF decode errors in buffer. Don't call the callback when the string is null.

Reply:
Thank you. I didn't consider that case. I will modify the code to avoid calling the callback if jMsg is nullptr.